### PR TITLE
Add --image-type option to GKE kube-up

### PR DESCRIPTION
**Pull request by wonderfly**
@maisem @roberthbailey Can you review?

cc/ @andyzheng0831

### CODE REVIEW

**Comment by roberthbailey** [comment]
FYI, all e2es are currently failing.

File: cluster/gke/util.sh [hunk 1]
```diff
@@ -166,6 +166,10 @@ function kube-up() {
     create_args+=("--additional-zones=${ADDITIONAL_ZONES}")
   fi
 
+  if [[ ! -z "${IMAGE_FAMILY}:-" ]]; then
```
Comment (brendandburns) [inline cluster/gke/util.sh]: I think that you mean `${IMAGE_FAMILY:-}` (the `:-` is currently after the curly brace.  Once you fix that LGTM.

Comment (wonderfly) [inline cluster/gke/util.sh]: Done

File: cluster/gke/config-test.sh [hunk 2]
```diff
@@ -18,6 +18,7 @@
 CLUSTER_NAME="${CLUSTER_NAME:-${USER}-gke-e2e}"
 NETWORK=${KUBE_GKE_NETWORK:-e2e}
 NODE_TAG="k8s-${CLUSTER_NAME}-node"
+IMAGE_TYPE="${KUBE_GKE_IMAGE_TYPE:-}"
```
Comment (zmerlynn) [inline cluster/gke/config-test.sh]: `IMAGE_TYPE` here as well? Then self-apply LGTM.

Comment (wonderfly) [inline cluster/gke/config-test.sh]: I've been following the convention that other env vars are defined: test overrides are always prefixed with `KUBE`. For example: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/config-test.sh

Also, I don't have the permission to add labels.

Comment (zmerlynn) [inline cluster/gke/config-test.sh]: Huh. That's odd and exceedingly patchwork.

Comment (zmerlynn) [inline cluster/gke/config-test.sh]: It's pretty random which variables seem to follow it and which don't.

LGTM.

**Comment by wonderfly** [comment]
@roberthbailey Is it able to cherry-pick this into 1.2 so that we can get GKE pinned e2e tests start running GCI on nodes as well?

cc/ @andyzheng0831 @kubernetes/goog-image

**Comment by andyzheng0831** [comment]
I think we should cherry pick it into release-1.2 branch. That will be useful for GCI team to test the GKE nodes of 1.2.X version kubelet on GCI.

**Pull request merged**